### PR TITLE
 betterbird: 102.8.0-bb30 -> 102.12.0-bb37 

### LINKED ISSUE #239531: Update request: betterbird 102.8.0-bb30 → 102.12.0-bb37
- Package name: betterbird
- Latest released version: 102.12.0-bb37
- Current version on the unstable channel: 102.8.0-bb30
- Current version on the stable/release channel: 102.8.0-bb30

- [ x ] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@SuperSandro2000 

The package is outdated since a while. I hope it can be updated in the near feature. Thank you :)

-----

Note for maintainers: Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Description of changes

Closes #239531

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/mailreaders/betterbird/default.nix [hunk 1]
```diff
@@ -9,54 +9,60 @@
...
+  };
+in ((buildMozillaMach {
   pname = "betterbird";
-  version = "102.8.0";
+  inherit version;
 
   applicationName = "Betterbird";
   binaryName = "betterbird";
-  inherit (thunderbird-unwrapped) application extraPatches;
+  inherit (thunderbird-unwrapped) application; # extraPatches # uncomment when nixpkgs thunderbird matches latest betterbird release again
```
Comment (mweinelt) [inline pkgs/applications/networking/mailreaders/betterbird/default.nix:45]: That is the case.

```
nix-repl> thunderbird-unwrapped.version
"102.12.0"

**Comment by emilazy** [comment]
Do you feel able to commit to trying to keep the version in all supported branches up to date with security fixes (at least those releases based on versions mentioned in the upstream [security advisories list](https://www.mozilla.org/en-US/security/known-vulnerabilities/thunderbird/)) going forward, or else set `knownVulnerabilities` in the relevant branches (including unstable if applicable)?

**Comment by SuperSandro2000** [comment]
I can update it to the versions released on [their website](https://www.betterbird.eu/downloads/index.php). 

> or else set `knownVulnerabilities` in the relevant branches (including unstable if applicable)?

No, we can't do that because it would remove the cache from cache.nixos.org and building thunderbird from source is not viable unless you have a high core server, essentially making it unusable for the average person.

**Comment by emilazy** [comment]
The average person shouldn't be using a version with known security vulnerabilities.

**Comment by SebTM** [comment]
```
❯ nix log /nix/store/a77ki5sdw7aqplzvpkdnjn6ciql0ylpb-betterbird-unwrapped-102.12.0.drv
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/a77ki5sdw7aqplzvpkdnjn6ciql0ylpb-betterbird-unwrapped-102.12.0.drv^*'
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/p1waky96ksydn57a71vgazwvhf5khiq7-thunderbird-102.12.0.source.tar.xz
source root is thunderbird-102.12.0
setting SOURCE_DATE_EPOCH to timestamp 1685848158 of file thunderbird-102.12.0/sourcestamp.txt
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
applying patch /nix/store/3qgdrdf8faarkk1q2l3xmnqpkaw37n4q-avoid-redefinition.patch?id=2f620d205ed0f9072bbd7714b5ec1b7bf6911c12
patching file dom/media/webrtc/transport/third_party/nICEr/src/stun/addrs-netlink.c
applying patch /nix/store/11g7gcb3cls6znviazm4553w6ns04sbz-bindgen.patch?id=4c4b0c01c808657fffc5b796c56108c57301b28f
patching file third_party/rust/bindgen/.cargo-checksum.json
patching file third_party/rust/bindgen/src/clang.rs
patching file third_party/rust/bindgen/src/ir/comp.rs
patching file third_party/rust/bindgen/src/ir/ty.rs
patching file third_party/rust/bindgen/src/ir/var.rs
applying patch /nix/store/i1132bnafs0grmj0mnmsmcg65dhcz83g-mp4parse-rust-170.patch
patching file third_party/rust/mp4parse/.cargo-checksum.json
patching file third_party/rust/mp4parse/src/lib.rs
applying patch /nix/store/2x0h67cqxa7bnffb63q2scly64rb2nlp-env_var_for_system_dir-ff86.patch
patching file toolkit/xre/nsXREDirProvider.cpp
Hunk #2 succeeded at 294 (offset -12 lines).
applying patch /nix/store/7zf2qf5fkyb5pv9yzg89a9rzrcffhzaw-no-buildconfig-ffx96.patch
patching file docshell/base/nsAboutRedirector.cpp
Hunk #1 succeeded at 73 (offset 7 lines).
patching file toolkit/content/jar.mn
applying patch /nix/store/0w4k77gky1ffyawxfch5s8bbxhriklwy-no-buildconfig-115.patch
patching file comm/mail/base/jar.mn
Reversed (or previously applied) patch detected!  Assume -R? [n]
Apply anyway? [n]
Skipping patch.
1 out of 1 hunk ignored -- saving rejects to file comm/mail/base/jar.mn.rej
```

This seems to have broken Betterbird on unstable? If I revert the commit it starts building the old version (which I canceled) as I don't have enough resources. Using the version from stable via overlay works for me.

**Comment by SuperSandro2000** [comment]
That's because betterbird uses thunderbird which now points at 115 but we still need 102. #243529

**Pull request merged**